<a href="https://colab.research.google.com/github/46470759/SQL-PROJECTS/blob/main/Logical_and_comparison_operators_ii.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Logical and comparison operators ii


## Learning objectives

In this train we will learn:
- How to use the IS NULL and IS NOT NULL statements
- How to use the IS IN and IS NOT IN statements
- Understand if there is any correlation between the GDP and availibility of drinking water and sanitation services in Sub-Saharan Africa

## Connecting to our MySQL database

Using our `Access_to_Basic_Services` table in our `united_nations` database we created in MySQL Workbench, we want to answer some questions about our dataset. We can apply the same queries we used in MySQL Workbench in this notebook if we connect to our MySQL server by running the cells below.


In [ ]:
# Load and activate the SQL extension to allows us to execute SQL in a Jupyter notebook.
# If you get an error here, make sure that mysql and pymysql is installed correctly.

%load_ext sql

In [ ]:
# Establish a connection to the local database using the '%sql' magic command,
# Replace 'password' with our connection password and `db_name` with our database name.
# If you get an error here, please make sure the database name or password is correct.

%sql mysql+pymysql://root:oshukumogbo9592@localhost:3306/united_nations

'Connected: root@united_nations'


To make a query, we add the `%%sql` command to the start of a cell, create one open line and then the query like below and run the cell.

In [ ]:
%%sql

SELECT
    *
FROM
    Access_to_Basic_Services
LIMIT 5;

 * mysql+pymysql://root:***@localhost:3306/united_nations
5 rows affected.


[('Central and Southern Asia', 'Central Asia', 'Kazakhstan', 2015, Decimal('94.67'), Decimal('98.00'), Decimal('17.542806'), Decimal('184.39'), Decimal('2699700.00'), Decimal('4.93')),
 ('Central and Southern Asia', 'Central Asia', 'Kazakhstan', 2016, Decimal('94.67'), Decimal('98.00'), Decimal('17.794055'), Decimal('137.28'), Decimal('2699700.00'), Decimal('4.96')),
 ('Central and Southern Asia', 'Central Asia', 'Kazakhstan', 2017, Decimal('95.00'), Decimal('98.00'), Decimal('18.037776'), Decimal('166.81'), Decimal('2699700.00'), Decimal('4.90')),
 ('Central and Southern Asia', 'Central Asia', 'Kazakhstan', 2018, Decimal('95.00'), Decimal('98.00'), Decimal('18.276452'), Decimal('179.34'), Decimal('2699700.00'), Decimal('4.85')),
 ('Central and Southern Asia', 'Central Asia', 'Kazakhstan', 2019, Decimal('95.00'), Decimal('98.00'), Decimal('18.513673'), Decimal('181.67'), Decimal('2699700.00'), Decimal('4.80'))]

## Exercise


We will be working with the `united_nations.Access_to_Basic_Services` table, which contains information about different countries, their access to basic services, and their estimated GDP.

In this exercise We would like to determine if the GDP of a country, specifically in Sub-Saharan Africa has any correlation to its access to basic services


### Task 1
Select data from the Sub-Saharan African region during the year 2020.

In [ ]:
%%sql

SELECT
    Country_name,
    Time_period,
    Pct_managed_drinking_water_services,
    Pct_managed_sanitation_services,
    Est_gdp_in_billions,
    region
FROM
    Access_to_Basic_Services
WHERE
        Region = 'Sub-Saharan Africa'
AND
    Time_period = 2020
LIMIT 20;

 * mysql+pymysql://root:***@localhost:3306/united_nations
20 rows affected.


[('Burundi', 2020, Decimal('70.33'), Decimal('44.33'), Decimal('2.65'), 'Sub-Saharan Africa'),
 ('Djibouti', 2020, Decimal('69.00'), Decimal('56.00'), Decimal('3.18'), 'Sub-Saharan Africa'),
 ('Ethiopia', 2020, Decimal('58.00'), Decimal('11.67'), Decimal('107.66'), 'Sub-Saharan Africa'),
 ('Kenya', 2020, Decimal('67.00'), Decimal('33.67'), Decimal('100.67'), 'Sub-Saharan Africa'),
 ('Madagascar', 2020, Decimal('56.33'), Decimal('13.00'), Decimal('13.05'), 'Sub-Saharan Africa'),
 ('Malawi', 2020, Decimal('74.33'), Decimal('28.67'), Decimal('12.18'), 'Sub-Saharan Africa'),
 ('Mauritius', 2020, Decimal('100.00'), Decimal('96.00'), Decimal('11.40'), 'Sub-Saharan Africa'),
 ('Mayotte', 2020, Decimal('96.00'), Decimal('100.00'), None, 'Sub-Saharan Africa'),
 ('Mozambique', 2020, Decimal('66.67'), Decimal('40.33'), Decimal('14.03'), 'Sub-Saharan Africa'),
 ('Rwanda', 2020, Decimal('66.33'), Decimal('64.00'), Decimal('10.18'), 'Sub-Saharan Africa'),
 ('Réunion', 2020, Decimal('100.00'), Decimal('100.00'), None, 'Sub-Saharan Africa'),
 ('Somalia', 2020, Decimal('57.33'), Decimal('40.00'), Decimal('6.88'), 'Sub-Saharan Africa'),
 ('South Sudan', 2020, Decimal('48.33'), Decimal('22.33'), None, 'Sub-Saharan Africa'),
 ('Uganda', 2020, Decimal('61.00'), Decimal('21.67'), Decimal('37.60'), 'Sub-Saharan Africa'),
 ('United Republic of Tanzania', 2020, Decimal('65.00'), Decimal('34.00'), None, 'Sub-Saharan Africa'),
 ('Zambia', 2020, Decimal('66.67'), Decimal('32.67'), Decimal('18.11'), 'Sub-Saharan Africa'),
 ('Zimbabwe', 2020, Decimal('68.00'), Decimal('36.33'), Decimal('21.51'), 'Sub-Saharan Africa'),
 ('Angola', 2020, Decimal('52.33'), Decimal('47.00'), Decimal('53.62'), 'Sub-Saharan Africa'),
 ('Cameroon', 2020, Decimal('64.00'), Decimal('43.00'), Decimal('40.77'), 'Sub-Saharan Africa'),
 ('Central African Republic', 2020, Decimal('38.33'), Decimal('15.00'), Decimal('2.33'), 'Sub-Saharan Africa')]

### Task 2

Sometimes there are null values in our entries. Any country having Null values for their GDP should not be included in our query as they will not help us determine if there is any correlation between GDP and access to basic services. For this task determine if there are any NULL values in the GDP column

In [ ]:
%%sql

SELECT
    Country_name,
    Time_period,
    Pct_managed_drinking_water_services,
    Pct_managed_sanitation_services,
    Est_gdp_in_billions,
    region
FROM
    Access_to_Basic_Services
WHERE
        Region = 'Sub-Saharan Africa'
AND
    Time_period = 2020
AND
    Est_gdp_in_billions IS NULL;

 * mysql+pymysql://root:***@localhost:3306/united_nations
9 rows affected.


[('Mayotte', 2020, Decimal('96.00'), Decimal('100.00'), None, 'Sub-Saharan Africa'),
 ('Réunion', 2020, Decimal('100.00'), Decimal('100.00'), None, 'Sub-Saharan Africa'),
 ('South Sudan', 2020, Decimal('48.33'), Decimal('22.33'), None, 'Sub-Saharan Africa'),
 ('United Republic of Tanzania', 2020, Decimal('65.00'), Decimal('34.00'), None, 'Sub-Saharan Africa'),
 ('Congo', 2020, Decimal('69.00'), Decimal('17.67'), None, 'Sub-Saharan Africa'),
 ('Democratic Republic of the Congo', 2020, Decimal('47.67'), Decimal('15.33'), None, 'Sub-Saharan Africa'),
 ("Côte d'Ivoire", 2020, Decimal('70.67'), Decimal('34.67'), None, 'Sub-Saharan Africa'),
 ('Gambia', 2020, Decimal('79.33'), Decimal('44.33'), None, 'Sub-Saharan Africa'),
 ('Saint Helena', 2020, Decimal('99.00'), Decimal('100.00'), None, 'Sub-Saharan Africa')]

### Task 3

If there are any Null values exclude them from your query.

In [ ]:
%%sql

SELECT
    Country_name,
    Time_period,
    Pct_managed_drinking_water_services,
    Pct_managed_sanitation_services,
    Est_gdp_in_billions,
    region
FROM
    Access_to_Basic_Services
WHERE
    Region = 'Sub-Saharan Africa'
AND
    Time_period = 2020
AND
    Est_gdp_in_billions IS NOT NULL;

 * mysql+pymysql://root:***@localhost:3306/united_nations
38 rows affected.


[('Burundi', 2020, Decimal('70.33'), Decimal('44.33'), Decimal('2.65'), 'Sub-Saharan Africa'),
 ('Djibouti', 2020, Decimal('69.00'), Decimal('56.00'), Decimal('3.18'), 'Sub-Saharan Africa'),
 ('Ethiopia', 2020, Decimal('58.00'), Decimal('11.67'), Decimal('107.66'), 'Sub-Saharan Africa'),
 ('Kenya', 2020, Decimal('67.00'), Decimal('33.67'), Decimal('100.67'), 'Sub-Saharan Africa'),
 ('Madagascar', 2020, Decimal('56.33'), Decimal('13.00'), Decimal('13.05'), 'Sub-Saharan Africa'),
 ('Malawi', 2020, Decimal('74.33'), Decimal('28.67'), Decimal('12.18'), 'Sub-Saharan Africa'),
 ('Mauritius', 2020, Decimal('100.00'), Decimal('96.00'), Decimal('11.40'), 'Sub-Saharan Africa'),
 ('Mozambique', 2020, Decimal('66.67'), Decimal('40.33'), Decimal('14.03'), 'Sub-Saharan Africa'),
 ('Rwanda', 2020, Decimal('66.33'), Decimal('64.00'), Decimal('10.18'), 'Sub-Saharan Africa'),
 ('Somalia', 2020, Decimal('57.33'), Decimal('40.00'), Decimal('6.88'), 'Sub-Saharan Africa'),
 ('Uganda', 2020, Decimal('61.00'), Decimal('21.67'), Decimal('37.60'), 'Sub-Saharan Africa'),
 ('Zambia', 2020, Decimal('66.67'), Decimal('32.67'), Decimal('18.11'), 'Sub-Saharan Africa'),
 ('Zimbabwe', 2020, Decimal('68.00'), Decimal('36.33'), Decimal('21.51'), 'Sub-Saharan Africa'),
 ('Angola', 2020, Decimal('52.33'), Decimal('47.00'), Decimal('53.62'), 'Sub-Saharan Africa'),
 ('Cameroon', 2020, Decimal('64.00'), Decimal('43.00'), Decimal('40.77'), 'Sub-Saharan Africa'),
 ('Central African Republic', 2020, Decimal('38.33'), Decimal('15.00'), Decimal('2.33'), 'Sub-Saharan Africa'),
 ('Chad', 2020, Decimal('52.67'), Decimal('18.67'), Decimal('10.72'), 'Sub-Saharan Africa'),
 ('Gabon', 2020, Decimal('73.33'), Decimal('47.00'), Decimal('15.31'), 'Sub-Saharan Africa'),
 ('Sao Tome and Principe', 2020, Decimal('77.33'), Decimal('46.00'), Decimal('0.47'), 'Sub-Saharan Africa'),
 ('Botswana', 2020, Decimal('89.67'), Decimal('74.33'), Decimal('14.93'), 'Sub-Saharan Africa'),
 ('Eswatini', 2020, Decimal('76.67'), Decimal('61.33'), Decimal('3.98'), 'Sub-Saharan Africa'),
 ('Lesotho', 2020, Decimal('76.33'), Decimal('49.67'), Decimal('2.23'), 'Sub-Saharan Africa'),
 ('Namibia', 2020, Decimal('83.67'), Decimal('35.00'), Decimal('10.58'), 'Sub-Saharan Africa'),
 ('South Africa', 2020, Decimal('92.00'), Decimal('78.67'), Decimal('337.62'), 'Sub-Saharan Africa'),
 ('Benin', 2020, Decimal('65.33'), Decimal('17.33'), Decimal('15.65'), 'Sub-Saharan Africa'),
 ('Burkina Faso', 2020, Decimal('53.33'), Decimal('25.00'), Decimal('17.93'), 'Sub-Saharan Africa'),
 ('Cabo Verde', 2020, Decimal('87.33'), Decimal('78.00'), Decimal('1.70'), 'Sub-Saharan Africa'),
 ('Ghana', 2020, Decimal('84.67'), Decimal('23.00'), Decimal('70.04'), 'Sub-Saharan Africa'),
 ('Guinea', 2020, Decimal('67.33'), Decimal('32.33'), Decimal('14.18'), 'Sub-Saharan Africa'),
 ('Guinea-Bissau', 2020, Decimal('60.00'), Decimal('19.33'), Decimal('1.43'), 'Sub-Saharan Africa'),
 ('Liberia', 2020, Decimal('75.00'), Decimal('17.67'), Decimal('3.04'), 'Sub-Saharan Africa'),
 ('Mali', 2020, Decimal('83.67'), Decimal('46.00'), Decimal('17.47'), 'Sub-Saharan Africa'),
 ('Mauritania', 2020, Decimal('70.33'), Decimal('48.00'), Decimal('8.41'), 'Sub-Saharan Africa'),
 ('Niger', 2020, Decimal('57.33'), Decimal('24.67'), Decimal('13.74'), 'Sub-Saharan Africa'),
 ('Nigeria', 2020, Decimal('77.33'), Decimal('42.67'), Decimal('432.20'), 'Sub-Saharan Africa'),
 ('Senegal', 2020, Decimal('85.00'), Decimal('57.00'), Decimal('24.49'), 'Sub-Saharan Africa'),
 ('Sierra Leone', 2020, Decimal('65.00'), Decimal('17.33'), Decimal('4.06'), 'Sub-Saharan Africa'),
 ('Togo', 2020, Decimal('70.67'), Decimal('20.00'), Decimal('7.57'), 'Sub-Saharan Africa')]

### Task 4

Lets get an idea if there's any correlation between the GDP and access to basic services for the top 5 economies in Sub-Saharan Africr. The top 5 GDP's are : ('Nigeria','South Africa','Ethiopia','Kenya','Ghana'). Make sure your query only includes these countries.

In [ ]:
%%sql

SELECT
    Country_name,
    Time_period,
    Pct_managed_drinking_water_services,
    Pct_managed_sanitation_services,
    Est_gdp_in_billions,
    region
FROM
    Access_to_Basic_Services
WHERE
    Region = 'Sub-Saharan Africa'
AND
    Time_period = 2020
AND
    Est_gdp_in_billions IS NOT NULL
AND
    Country_name
    IN (
'Nigeria','South Africa','Ethiopia','Kenya','Ghana'
    );


 * mysql+pymysql://root:***@localhost:3306/united_nations
5 rows affected.


[('Ethiopia', 2020, Decimal('58.00'), Decimal('11.67'), Decimal('107.66'), 'Sub-Saharan Africa'),
 ('Kenya', 2020, Decimal('67.00'), Decimal('33.67'), Decimal('100.67'), 'Sub-Saharan Africa'),
 ('South Africa', 2020, Decimal('92.00'), Decimal('78.67'), Decimal('337.62'), 'Sub-Saharan Africa'),
 ('Ghana', 2020, Decimal('84.67'), Decimal('23.00'), Decimal('70.04'), 'Sub-Saharan Africa'),
 ('Nigeria', 2020, Decimal('77.33'), Decimal('42.67'), Decimal('432.20'), 'Sub-Saharan Africa')]

Based on your results. Is there any correlation between the GDP and access to basic services? Does a higher GDP translate to better services?

Answer: based on the result, there is no correlation between GDP and access to basic services as a higher GDP deos not necessarily translate to better services.

### Task 5

We only looked at 5 countries in the previous query. Lets have a look at the rest of Sub-Saharan Africa .
Exclude the countries mentioned in the previous task

In [ ]:
%%sql


SELECT
    Country_name,
    Time_period,
    Pct_managed_drinking_water_services,
    Pct_managed_sanitation_services,
    Est_gdp_in_billions,
    region
FROM
    Access_to_Basic_Services
WHERE
    Region = 'Sub-Saharan Africa'
AND
    Time_period = 2020
AND
    Est_gdp_in_billions IS NOT NULL
AND
    Country_name
    NOT IN (
'Nigeria','South Africa','Ethiopia','Kenya','Ghana'
    )
ORDER BY
    Pct_managed_drinking_water_services;

 * mysql+pymysql://root:***@localhost:3306/united_nations
33 rows affected.


[('Central African Republic', 2020, Decimal('38.33'), Decimal('15.00'), Decimal('2.33'), 'Sub-Saharan Africa'),
 ('Angola', 2020, Decimal('52.33'), Decimal('47.00'), Decimal('53.62'), 'Sub-Saharan Africa'),
 ('Chad', 2020, Decimal('52.67'), Decimal('18.67'), Decimal('10.72'), 'Sub-Saharan Africa'),
 ('Burkina Faso', 2020, Decimal('53.33'), Decimal('25.00'), Decimal('17.93'), 'Sub-Saharan Africa'),
 ('Madagascar', 2020, Decimal('56.33'), Decimal('13.00'), Decimal('13.05'), 'Sub-Saharan Africa'),
 ('Somalia', 2020, Decimal('57.33'), Decimal('40.00'), Decimal('6.88'), 'Sub-Saharan Africa'),
 ('Niger', 2020, Decimal('57.33'), Decimal('24.67'), Decimal('13.74'), 'Sub-Saharan Africa'),
 ('Guinea-Bissau', 2020, Decimal('60.00'), Decimal('19.33'), Decimal('1.43'), 'Sub-Saharan Africa'),
 ('Uganda', 2020, Decimal('61.00'), Decimal('21.67'), Decimal('37.60'), 'Sub-Saharan Africa'),
 ('Cameroon', 2020, Decimal('64.00'), Decimal('43.00'), Decimal('40.77'), 'Sub-Saharan Africa'),
 ('Sierra Leone', 2020, Decimal('65.00'), Decimal('17.33'), Decimal('4.06'), 'Sub-Saharan Africa'),
 ('Benin', 2020, Decimal('65.33'), Decimal('17.33'), Decimal('15.65'), 'Sub-Saharan Africa'),
 ('Rwanda', 2020, Decimal('66.33'), Decimal('64.00'), Decimal('10.18'), 'Sub-Saharan Africa'),
 ('Mozambique', 2020, Decimal('66.67'), Decimal('40.33'), Decimal('14.03'), 'Sub-Saharan Africa'),
 ('Zambia', 2020, Decimal('66.67'), Decimal('32.67'), Decimal('18.11'), 'Sub-Saharan Africa'),
 ('Guinea', 2020, Decimal('67.33'), Decimal('32.33'), Decimal('14.18'), 'Sub-Saharan Africa'),
 ('Zimbabwe', 2020, Decimal('68.00'), Decimal('36.33'), Decimal('21.51'), 'Sub-Saharan Africa'),
 ('Djibouti', 2020, Decimal('69.00'), Decimal('56.00'), Decimal('3.18'), 'Sub-Saharan Africa'),
 ('Burundi', 2020, Decimal('70.33'), Decimal('44.33'), Decimal('2.65'), 'Sub-Saharan Africa'),
 ('Mauritania', 2020, Decimal('70.33'), Decimal('48.00'), Decimal('8.41'), 'Sub-Saharan Africa'),
 ('Togo', 2020, Decimal('70.67'), Decimal('20.00'), Decimal('7.57'), 'Sub-Saharan Africa'),
 ('Gabon', 2020, Decimal('73.33'), Decimal('47.00'), Decimal('15.31'), 'Sub-Saharan Africa'),
 ('Malawi', 2020, Decimal('74.33'), Decimal('28.67'), Decimal('12.18'), 'Sub-Saharan Africa'),
 ('Liberia', 2020, Decimal('75.00'), Decimal('17.67'), Decimal('3.04'), 'Sub-Saharan Africa'),
 ('Lesotho', 2020, Decimal('76.33'), Decimal('49.67'), Decimal('2.23'), 'Sub-Saharan Africa'),
 ('Eswatini', 2020, Decimal('76.67'), Decimal('61.33'), Decimal('3.98'), 'Sub-Saharan Africa'),
 ('Sao Tome and Principe', 2020, Decimal('77.33'), Decimal('46.00'), Decimal('0.47'), 'Sub-Saharan Africa'),
 ('Mali', 2020, Decimal('83.67'), Decimal('46.00'), Decimal('17.47'), 'Sub-Saharan Africa'),
 ('Namibia', 2020, Decimal('83.67'), Decimal('35.00'), Decimal('10.58'), 'Sub-Saharan Africa'),
 ('Senegal', 2020, Decimal('85.00'), Decimal('57.00'), Decimal('24.49'), 'Sub-Saharan Africa'),
 ('Cabo Verde', 2020, Decimal('87.33'), Decimal('78.00'), Decimal('1.70'), 'Sub-Saharan Africa'),
 ('Botswana', 2020, Decimal('89.67'), Decimal('74.33'), Decimal('14.93'), 'Sub-Saharan Africa'),
 ('Mauritius', 2020, Decimal('100.00'), Decimal('96.00'), Decimal('11.40'), 'Sub-Saharan Africa')]

In [ ]:
%%sql


SELECT
    Country_name,
    Time_period,
    Pct_managed_drinking_water_services,
    Pct_managed_sanitation_services,
    Est_gdp_in_billions,
    region
FROM
    Access_to_Basic_Services
WHERE
    Region = 'Sub-Saharan Africa'
AND
    Time_period = 2020
AND
    Est_gdp_in_billions IS NOT NULL
AND
    Country_name
    NOT IN (
'Nigeria','South Africa','Ethiopia','Kenya','Ghana'
    )
ORDER BY
    Pct_managed_sanitation_services;

 * mysql+pymysql://root:***@localhost:3306/united_nations
33 rows affected.


[('Madagascar', 2020, Decimal('56.33'), Decimal('13.00'), Decimal('13.05'), 'Sub-Saharan Africa'),
 ('Central African Republic', 2020, Decimal('38.33'), Decimal('15.00'), Decimal('2.33'), 'Sub-Saharan Africa'),
 ('Benin', 2020, Decimal('65.33'), Decimal('17.33'), Decimal('15.65'), 'Sub-Saharan Africa'),
 ('Sierra Leone', 2020, Decimal('65.00'), Decimal('17.33'), Decimal('4.06'), 'Sub-Saharan Africa'),
 ('Liberia', 2020, Decimal('75.00'), Decimal('17.67'), Decimal('3.04'), 'Sub-Saharan Africa'),
 ('Chad', 2020, Decimal('52.67'), Decimal('18.67'), Decimal('10.72'), 'Sub-Saharan Africa'),
 ('Guinea-Bissau', 2020, Decimal('60.00'), Decimal('19.33'), Decimal('1.43'), 'Sub-Saharan Africa'),
 ('Togo', 2020, Decimal('70.67'), Decimal('20.00'), Decimal('7.57'), 'Sub-Saharan Africa'),
 ('Uganda', 2020, Decimal('61.00'), Decimal('21.67'), Decimal('37.60'), 'Sub-Saharan Africa'),
 ('Niger', 2020, Decimal('57.33'), Decimal('24.67'), Decimal('13.74'), 'Sub-Saharan Africa'),
 ('Burkina Faso', 2020, Decimal('53.33'), Decimal('25.00'), Decimal('17.93'), 'Sub-Saharan Africa'),
 ('Malawi', 2020, Decimal('74.33'), Decimal('28.67'), Decimal('12.18'), 'Sub-Saharan Africa'),
 ('Guinea', 2020, Decimal('67.33'), Decimal('32.33'), Decimal('14.18'), 'Sub-Saharan Africa'),
 ('Zambia', 2020, Decimal('66.67'), Decimal('32.67'), Decimal('18.11'), 'Sub-Saharan Africa'),
 ('Namibia', 2020, Decimal('83.67'), Decimal('35.00'), Decimal('10.58'), 'Sub-Saharan Africa'),
 ('Zimbabwe', 2020, Decimal('68.00'), Decimal('36.33'), Decimal('21.51'), 'Sub-Saharan Africa'),
 ('Somalia', 2020, Decimal('57.33'), Decimal('40.00'), Decimal('6.88'), 'Sub-Saharan Africa'),
 ('Mozambique', 2020, Decimal('66.67'), Decimal('40.33'), Decimal('14.03'), 'Sub-Saharan Africa'),
 ('Cameroon', 2020, Decimal('64.00'), Decimal('43.00'), Decimal('40.77'), 'Sub-Saharan Africa'),
 ('Burundi', 2020, Decimal('70.33'), Decimal('44.33'), Decimal('2.65'), 'Sub-Saharan Africa'),
 ('Mali', 2020, Decimal('83.67'), Decimal('46.00'), Decimal('17.47'), 'Sub-Saharan Africa'),
 ('Sao Tome and Principe', 2020, Decimal('77.33'), Decimal('46.00'), Decimal('0.47'), 'Sub-Saharan Africa'),
 ('Gabon', 2020, Decimal('73.33'), Decimal('47.00'), Decimal('15.31'), 'Sub-Saharan Africa'),
 ('Angola', 2020, Decimal('52.33'), Decimal('47.00'), Decimal('53.62'), 'Sub-Saharan Africa'),
 ('Mauritania', 2020, Decimal('70.33'), Decimal('48.00'), Decimal('8.41'), 'Sub-Saharan Africa'),
 ('Lesotho', 2020, Decimal('76.33'), Decimal('49.67'), Decimal('2.23'), 'Sub-Saharan Africa'),
 ('Djibouti', 2020, Decimal('69.00'), Decimal('56.00'), Decimal('3.18'), 'Sub-Saharan Africa'),
 ('Senegal', 2020, Decimal('85.00'), Decimal('57.00'), Decimal('24.49'), 'Sub-Saharan Africa'),
 ('Eswatini', 2020, Decimal('76.67'), Decimal('61.33'), Decimal('3.98'), 'Sub-Saharan Africa'),
 ('Rwanda', 2020, Decimal('66.33'), Decimal('64.00'), Decimal('10.18'), 'Sub-Saharan Africa'),
 ('Botswana', 2020, Decimal('89.67'), Decimal('74.33'), Decimal('14.93'), 'Sub-Saharan Africa'),
 ('Cabo Verde', 2020, Decimal('87.33'), Decimal('78.00'), Decimal('1.70'), 'Sub-Saharan Africa'),
 ('Mauritius', 2020, Decimal('100.00'), Decimal('96.00'), Decimal('11.40'), 'Sub-Saharan Africa')]

Again. Do you see any correlation? Arrange the `Pct_managed_drinking_water_services` in descending order. Do the countries with the highest percentage to `Pct_managed_drinking_water_services` have a higher GDP then the countries with a lower percentage of `Pct_managed_drinking_water_services` ?
 Now arrange the `Pct_managed_sanitation_services` in descending order. Do countries with a higher `Pct_managed_sanitation_services` also have a higher GDP?

 Answer: there is no correlation.

## Solutions

### Task 1

In [ ]:
%%sql

SELECT
	Country_name,
	Time_period,
	Pct_managed_drinking_water_services,
	Pct_managed_sanitation_services,
	Est_population_in_millions,
	Est_gdp_in_billions
FROM
	united_nations.Access_to_Basic_Services
WHERE
	region = ‘Sub-Saharan Africa’
AND
    Time_period = 2020;

### Task 2

In [ ]:
%%sql

SELECT
	Country_name,
	Time_period,
	Pct_managed_drinking_water_services,
	Pct_managed_sanitation_services,
	Est_gdp_in_billions,
    region
FROM
	united_nations.Access_to_Basic_Services
WHERE
	region = 'Sub-Saharan Africa'
AND
	Time_period = 2020
AND
	Est_gdp_in_billions IS NULL;

# Use LIMIT if you think the results set will be large



### Task 3

In [ ]:
%%sql

SELECT
	Country_name,
	Time_period,
	Pct_managed_drinking_water_services,
	Pct_managed_sanitation_services,
	Est_gdp_in_billions,
    	region
FROM
	united_nations.Access_to_Basic_Services
WHERE
	region = 'Sub-Saharan Africa'
AND
	Time_period = 2020
AND
	Est_gdp_in_billions IS NOT NULL;

# Use LIMIT if you think the results set will be large
# Use ORDER BY Est_gdp_in_billions to order your results

### Task 4

Arrange the data in descending order. Comparing Nigeria to South Africa.  We see that Nigeria has a lower percentage in the availability of both water and sanitation services.


In [ ]:
%%sql

SELECT
	Country_name,
	Time_period,
	Pct_managed_drinking_water_services,
	Pct_managed_sanitation_services,
	Est_population_in_millions,
	Est_gdp_in_billions
FROM
	united_nations.Access_to_Basic_Services

WHERE
	Region = ‘Sub-Saharan Africa
AND
Time_period = 2020

AND
	Est_gdp_in_billions IS NOT NULL

AND
	Country_name  NOT IN ('Nigeria','South Africa','Ethiopia','Kenya','Ghana');


# Use LIMIT if you think the results set will be large
# Use ORDER BY Est_gdp_in_billions to order your results


### Task 5

Looking at the first entry without ordering any data it is interesting to note that Burundi, with a GDP of only 2.65 Billion has a similar percentage of managed services compared to Nigeria seen in our previous  query(77% and 42%). Nigeria had a percentage of 77 for managed water and 42 for managed sanitation. Arranging the percentage managed drinking water services in descending order.Mauritius with a GDP lower than that of Botswana has better drinking water and sanitation services.

In [ ]:
%%sql

SELECT
	Country_name,
	Time_period,
	Pct_managed_drinking_water_services,
	Pct_managed_sanitation_services,
	Est_population_in_millions,
	Est_gdp_in_billions
FROM
	united_nations.Access_to_Basic_Services

WHERE
	Region = ‘Sub-Saharan Africa
AND
Time_period = 2020

AND
	Est_gdp_in_billions IS NOT NULL

AND
	Country_name  IS IN ('Nigeria','South Africa','Ethiopia','Kenya','Ghana');

# Use LIMIT if you think the results set will be large
# Use ORDER BY Est_gdp_in_billions to order your results by GDP
# Use ORDER BY Pct_managed_drinking_water_services to order percentage managed drinking water services
# Use ORDER BY Pct_managed_sanitation_services to order your results by percentage managed sanitation services


## Summary

In this exercise we used the IS NULL statement to determine if there were any null values in the GDP column.
We then used the IS NOT NULL statement to exclude those nulls.
We looked at the top 5 GDP's in Sub-Saharan Africa by using the IS IN statement.
We had a look at the rest of the Sub-Saharan African countries by excluding the top 5 GDP's using the IS NOT IN statement
Based on the quick perusal of the data in Sub-Saharan africa, we can conclude that there isn’t
any noticeable correlation between GDP and the availability of drinking water and sanitation services.
